File: mobile/src/main/java/com/nomi/artwatch/di/component/ActivityComponent.java
package com.nomi.artwatch.di.component;

import com.nomi.artwatch.di.ActivityScope;
import com.nomi.artwatch.di.module.ActivityModule;
import com.nomi.artwatch.ui.activity.HistoryActivity;
import com.nomi.artwatch.ui.activity.LoginActivity;
import com.nomi.artwatch.ui.activity.MainActivity;
import com.nomi.artwatch.ui.activity.SettingActivity;
import com.nomi.artwatch.ui.activity.SplashActivity;

import dagger.Component;

/**
 * Created by Ryota Niinomi on 2015/11/03.
 */
@ActivityScope
@Component(
        dependencies = ApplicationComponent.class,
        modules = {
                ActivityModule.class
        }
)
public interface ActivityComponent {
    void inject(SplashActivity activity);
    void inject(LoginActivity activity);
    void inject(MainActivity activity);
    void inject(HistoryActivity activity);
    void inject(SettingActivity activity);
}

File: mobile/src/main/java/com/nomi/artwatch/di/component/ActivityComponent.kt
package com.nomi.artwatch.di.component

import com.nomi.artwatch.di.ActivityScope
import com.nomi.artwatch.di.module.ActivityModule
import com.nomi.artwatch.ui.activity.*
import dagger.Component

/**
 * Created by Ryota Niinomi on 2015/11/03.
 */
@ActivityScope
@Component(dependencies = arrayOf(ApplicationComponent::class), modules = arrayOf(ActivityModule::class))
interface ActivityComponent {
    fun inject(activity: SplashActivity)
    fun inject(activity: LoginActivity)
    fun inject(activity: MainActivity)
    fun inject(activity: HistoryActivity)
    fun inject(activity: SettingActivity)
    fun inject(activity: OssActivity)
}
